orttraining/orttraining/training_ops/cuda/optimizer/adamw/adamw.cc

```diff
@@ -35,7 +35,7 @@ Status AdamWOptimizer::ComputeInternal(OpKernelContext* ctx) const {
   AdamWOptimizerBase::Prepare p;
   ORT_RETURN_IF_ERROR(PrepareForCompute(ctx, p));
 
-  int64_t* updated_flag_ptr = p.updated_flag->template MutableData<int64_t>();
+  bool* updated_flag_ptr = p.updated_flag->template MutableData<bool>();
```

Comment:
Why is this change made?

Reply:
This change was needed because the code was originally checking if the output updated_flag was an integer, but in the SGD optimizer, it was defined as a boolean. Instead of adding a new check for booleans, I made the definition of adamW match the boolean type, and then changed the check to only look for booleans.

Reply:
This is a legacy todo, SGD and AdamW define updated_flag in different data types.

Reply:
@AdamLouly you probably need to set the `updated_flag_ptr` accordingly (to true or false instead of 1 and 0)